### PR TITLE
typescript sdk list function supports all arguments that the backend

### PR DESCRIPTION
**Pull request by DanielCollins**
understands

### CODE REVIEW

File: typescript/src/entities/product.test.ts [hunk 1]
```diff
@@ -109,14 +113,105 @@ test('can list products from server', () => {
   });
 });
 
+test('can list products with options set', () => {
+  const merchi = new Merchi();
+  const options = {embed: {},
```
Comment (s3341458) [inline typescript/src/entities/product.test.ts]: I personally like indented way to format dictionary, like

```
a = {
     b: 1
     c:  2
}
   

```

File: typescript/src/entity.ts [hunk 2]
```diff
@@ -149,8 +205,147 @@ export class Entity {
...
+        fetchOptions.query.push(['public_only', options.publicOnly.toString()]);
+      } 
+      if (options.managedOnly !== undefined) {
+        fetchOptions.query.push(['managed_only',
+          options.managedOnly.toString()]);
+      } 
+      if (options.memberOnly !== undefined) {
+        fetchOptions.query.push(['member_only', options.memberOnly.toString()]);
+      } 
+      if (options.domainRoles) {
```
Comment (s3341458) [inline typescript/src/entity.ts]: domainRoles does not need "options.domainRoles !== undefined"

Comment (s3341458) [reply typescript/src/entity.ts]: just want to mention in JS Boolean([]) equal to true, I do not think typescript changed that behaivor?

Comment (DanielCollins) [reply typescript/src/entity.ts]: the `[]` doesn't cause an observable bug, but you are correct, it should be more consisitent with the others

Comment (DanielCollins) [reply typescript/src/entity.ts]: `embed` and `withRights` are intentionally different

File: typescript/src/entities/product.test.ts (hunk 1 above)
Comment (DanielCollins) [reply typescript/src/entities/product.test.ts]: personally i hate both the style used here, and the style you've shown.

it's just a matter of whatever the current eslint config says. i should take a look at that, but not in this pr

**Pull request merged**